### PR TITLE
feat(agent/alloy): migrate from Grafana agent to Grafana Alloy

### PR DESCRIPTION
**Pull request by bubu11e**
Migrate from Grafana agent to Grafana Alloy

## Why
Grafana agent has been deprecated in favor of Grafana alloy.

## What
Replace Grafana agent by Grafana Alloy

## Links
  - https://grafana.com/blog/2024/04/09/grafana-agent-to-grafana-alloy-opentelemetry-collector-faq/
  - https://grafana.com/blog/2024/04/09/grafana-alloy-opentelemetry-collector-with-prometheus-pipelines/

## Checklist
- [X] Changelog updated
- [X] Documentation updated

### CODE REVIEW

File: docs/sources/tutorials/quick-start-browser.md [hunk 1]
```diff
@@ -5,71 +5,96 @@ This document describes how to set up and use Grafana Faro Web SDK. For more inf
...
+	level = "info"
+}
+
+faro.receiver "integrations_app_agent_receiver" {
+	server {
+		listen_address           = "0.0.0.0"
+		listen_port              = 12345
+		cors_allowed_origins     = ["*"]
+		api_key                  = my_super_app_key
+		max_allowed_payload_size = "47MiB700KiB128B"
```
Comment (codecapitano) [inline docs/sources/tutorials/quick-start-browser.md:57]: This is fairly big allow size. 
The [default in faro receiver is 5MB](https://grafana.com/docs/alloy/latest/reference/components/faro.receiver/#server-block).

Best ist to remove that setting from the example.

File: infra/alloy/config/config.alloy [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		app  = env("DEMO_CLIENT_PACKAGE_NAME"),
+		kind = "",
+	}
+
+	server {
+		listen_address           = "0.0.0.0"
+		listen_port              = env("AGENT_PORT_APP_RECEIVER")
+		cors_allowed_origins     = ["*"]
+		api_key                  = env("AGENT_KEY_APP_RECEIVER")
+		max_allowed_payload_size = "47MiB700KiB128B"
```
Comment (codecapitano) [inline infra/alloy/config/config.alloy:76]: This is fairly big allow size. 
The [default in faro receiver is 5MB](https://grafana.com/docs/alloy/latest/reference/components/faro.receiver/#server-block).

Let's remove that setting.

File: infra/alloy/config/config.alloy [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+		rate_limiting {
```
Comment (codecapitano) [inline infra/alloy/config/config.alloy:78]: Let's keep the [default rate of 50 req/sec](https://grafana.com/docs/alloy/latest/reference/components/faro.receiver/#server-block).

Out of curiosity, did you experience any issue with the default value?

File: infra/alloy/config/config.alloy [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+faro.receiver "integrations_app_agent_receiver" {
+	extra_log_labels = {
+		app  = env("DEMO_CLIENT_PACKAGE_NAME"),
+		kind = "",
+	}
+
+	server {
+		listen_address           = "0.0.0.0"
+		listen_port              = env("AGENT_PORT_APP_RECEIVER")
+		cors_allowed_origins     = ["*"]
```
Comment (codecapitano) [inline infra/alloy/config/config.alloy:74]: The demo app rund on port 5173.
Let's be more strict by explicitly allowing localhost:5173.

**Review by codecapitano — CHANGES_REQUESTED** [review]
Hi @bubu11e great contribution thank you so much for this 🥳 

Just some minor things just to have some stricter defaults for users just copying the code. 🙏

File: docs/sources/tutorials/quick-start-browser.md [hunk 5]
```diff
@@ -5,71 +5,96 @@ This document describes how to set up and use Grafana Faro Web SDK. For more inf
...
+      values = { "kind" = "kind", "service_name" = "service_name", "app" = "app" }
+  }
+}
+
+loki.write "logs_write_client" {
+	endpoint {
+		url = <loki_write_url>
+	}
+	external_labels = {
+        }
```
Comment (rlankfo) [inline docs/sources/tutorials/quick-start-browser.md:44]: is this necessary? is it possible this is dropping labels?

File: docs/sources/tutorials/quick-start-browser.md [hunk 6]
```diff
@@ -5,71 +5,96 @@ This document describes how to set up and use Grafana Faro Web SDK. For more inf
...
+
+logging {
+	level = "info"
+}
+
+faro.receiver "integrations_app_agent_receiver" {
+	server {
+		listen_address           = "0.0.0.0"
+		listen_port              = 12345
+		cors_allowed_origins     = ["*"]
```
[the remark this answers is not in the document]
Comment (bubu11e) [reply docs/sources/tutorials/quick-start-browser.md]: Good catch.

File: docs/sources/tutorials/quick-start-browser.md (hunk 1 above)
Comment (bubu11e) [reply docs/sources/tutorials/quick-start-browser.md]: You are right.

File: infra/alloy/config/config.alloy (hunk 2 above)
Comment (bubu11e) [reply infra/alloy/config/config.alloy]: Fixed.

File: infra/alloy/config/config.alloy (hunk 3 above)
Comment (bubu11e) [reply infra/alloy/config/config.alloy:78]: Hi, I did not encounter any issues just the result of the conversion configuration tool.

File: docs/sources/tutorials/quick-start-browser.md (hunk 5 above)
Comment (bubu11e) [reply docs/sources/tutorials/quick-start-browser.md]: You are absolutely right, this is no needed at all.

File: infra/alloy/config/config.alloy (hunk 4 above)
Comment (bubu11e) [reply infra/alloy/config/config.alloy:74]: I did not manage to get it to work with anything else than "*". Furthermore this was present in the previous configuration.

Comment (codecapitano) [reply infra/alloy/config/config.alloy:74]: Yeah that's fine and there's no need to invest too much time here 👍

File: infra/alloy/config/config.alloy (hunk 3 above)
Comment (codecapitano) [reply infra/alloy/config/config.alloy:78]: All right cool. Then lets us the default rate of 50 req/sec.

**Comment by codecapitano** [comment]
@bubu11e I see there are some linter issues which are a bit hard to find. 
If it is fine for you I can fix them for you.

Just give me a 👍 / 👎

**Comment by codecapitano** [comment]
> @bubu11e I see there are some linter issues which are a bit hard to find sometimes. If it is fine for you I can fix them for you.
> 
> Just give me a 👍 / 👎

All right @bubu11e linter issue is fixed 👍

**Comment by bubu11e** [comment]
> > @bubu11e I see there are some linter issues which are a bit hard to find sometimes. If it is fine for you I can fix them for you.
> > Just give me a 👍 / 👎
> 
> All right @bubu11e linter issue is fixed 👍

Thanks a lot.

File: docs/sources/tutorials/quick-start-browser.md [hunk 7]
```diff
@@ -5,71 +5,93 @@ This document describes how to set up and use Grafana Faro Web SDK. For more inf
...
+otelcol.exporter.otlp "trace_write" {
+    retry_on_failure {
+        max_elapsed_time = "1m0s"
+    }
+
+    client {
+        endpoint = <tempo_endpoint>
+
+        tls {
+            insecure = true
```
Comment (codecapitano) [inline docs/sources/tutorials/quick-start-browser.md:91]: @bubu11e I have a last one.

The [tls.insecure setting is optional](https://grafana.com/docs/alloy/latest/reference/components/otelcol.exporter.otlp/#tls-block) and we shouldn't set it to insecure by default.
User should explicitly decide for this setting if they need to.

Comment (bubu11e) [reply docs/sources/tutorials/quick-start-browser.md]: Would be better indeed. Thanks and Fixed.

**Review by codecapitano — APPROVED** [review]
Great contribution. 
This was quite often requested since alloy was released.

Thank you so much 🙏 🥳

**Pull request merged**